src/t_stream.c

```diff
@@ -1434,7 +1429,7 @@ void streamReplyWithCGLag(client *c, stream *s, streamCG *cg) {
         /* The lag of a newly-initialized stream is 0. */
         lag = 0;
         valid = 1;
-    } else if (cg->entries_read != SCG_INVALID_ENTRIES_READ && !streamRangeHasTombstones(s, &cg->last_id, NULL)) {
+    } else if (cg->entries_read != SCG_INVALID_ENTRIES_READ && streamCompareID(&cg->last_id, &s->first_id) >= 0 && !streamRangeHasTombstones(s, &cg->last_id, NULL)) {
```

Comment:
It's a very long line. Let's break it.

```suggestion
    } else if (cg->entries_read != SCG_INVALID_ENTRIES_READ &&
               streamCompareID(&cg->last_id, &s->first_id) >= 0 &&
               !streamRangeHasTombstones(s, &cg->last_id, NULL)) {
```